# 3 УРОК УСЛОВНАЯ ВЕРОЯТНОСТЬ

## Задание 1

Из 500 компьютеров  180 принадлежат к 1 партии, 170 – ко второй партии, остальные к третьей. В первой партии 3% брака, во второй — 2%, в третьей – 6%. Случайно выбирается один компьютер. Определить вероятность того, что выбранный компьютер – бракованный.  Округлите полученный ответ до десятитысячных. В качестве разделителя используйте точку.

Подсказки

In [ ]:
500 
180 - 1 - 3%
170 - 2 - 2%
150 - 3 - 6%

In [1]:
#общая вероятность брака
Pb = 180/500*0.03 + 170/500*0.02 + 150/500*0.06
Pb

0.03559999999999999

In [2]:
round(Pb, 4)

0.0356

- **Что надо иметь в виду при решении?**
    
    Здесь нам понадобится лишь частичка формулы Байеса. Вспомните, что:
    
    $$
    P(X) = P(Y_1)P(X|Y_1)+ P(Y_2)P(X|Y_2) +...+P(Y_n)P(X|Y_n)
    $$
    
    P(X) - это как раз вероятность брака компьютера. Остальное связано с вероятностями:
    
    - ...взять **случайный** компьютер из **конкретной** партии (1, 2 или 3)
    - ...получить **бракованный** компьютер, **если** он взят из **конкретной** партии (1, 2 или 3)

##### ОТВЕТ: 0.0356 с первой попытки без подсказок

**Разбор решения**

Давайте найдем вероятность каждого события по классической определении вероятности. Всего 500 компьютеров, из них первой партии принадлежат 180, то есть

$$
P(H1) = {180 \over 500} = 0.36,
$$

2-ой партии принадлежат 170, то есть

$$
P(H2) = {170 \over 500} = 0.34,
$$

остальные компьютеры принадлежат к третей партии, то есть 

$$
P(H3) = {150 \over 500} = 0.3
$$

Введем событие А = (бракованый компьютер). По условию даны априорные вероятности

$$
P(A|H1) = 0.03
$$

$$
P(A|H2) = 0.02
$$

$$
P(A|H3) = 0.06
$$

Вероятность события А можем найти по формуле полной вероятности:
$$
P(A) = P(A|H1)P(H1) + P(A|H2)P(H2) + P(A|H3)P(H3) = 0.03 \times 0.36 + 0.02 \times 0.34 + 0.06 \times 0.3 = 0.0356
$$

## Задание 2

Есть компания с программистами. В ней на языке Java пишут в 3 раза больше человек, чем на языке C++. Программистов на языке C++ в 4 раза больше, чем на Python. Вероятность сделать работу за один день для Java-программиста — 0,85, для программиста на языке C++ 0,9, а для Python программиста 0,8. Найти вероятность того, что программист, выбранный наугад, сделает работу за один день. Ответ округлите до сотых. В качестве десятичного разделителя используйте точку.

Подсказки

In [ ]:
Python                1 - 0.8
C++(в 4 раза больше)  4 - 0.9
Java(в 3 раза больше) 12 - 0.85
Всего выходит 17х

In [3]:
#Общая вероятность кто то сделает за день
Pa = 1/17*0.8 + 4/17*0.9 + 12/17*0.85
Pa

0.8588235294117647

In [7]:
round(Pa, 2)

0.86

In [5]:
# Это уже не к заданию
#вероятность что это Питонист 
1/17 
#это вероятность что выбранный прогер - питонист

0.058823529411764705

In [6]:
#Формула вероятности что решил Питонист
1/17 * 0.8 / 1/17*0.8 + 4/17*0.9 + 12/17*0.85

0.8139792387543252

- **Ну и как тут вычислить количество программистов?**
    
    Точное их количество знать необязательно. Достаточно принять число питонистов за $x$, ну а дальше:
    
    $$
    x + 4x+3*4x = 17x
    $$
    
    Соответственно, вероятность того, что случайно выбранный программист пользуется питоном, равна ${x \over 17x} = {1 \over17}$. По той же аналогии вычисляются остальные вероятности.
    
    Ну а дальше - тот же алгоритм, что и в прошлой задаче.

###### ОТВЕТ: 0.86 с первой попытки без подсказок

Давайте найдем вероятность того, на каком языке пишет каждый программист, где наименьшую долю выразим через x. Тогда программистов на языке Python - x, программистов на языке C++ - 4x, а програмистов на языке Java - 12x:

$$
P(H1) = {x \over x + 4x + 12x} = {1 \over 17} -Python
$$

$$
P(H2) = {4x \over x + 4x + 12x} = {4 \over 17} - C++
$$

$$
P(H3) = {12x \over x + 4x + 12x} = {12 \over 17} - Java
$$

Введем событие А = (успешно решенная задача). По условию даны априорные вероятности:

$$
P(A|H1) = 0.8
$$

$$
P(A|H2) = 0.9
$$

$$
P(A|H3) = 0.85
$$

Вероятность события А можем найти по формуле полной вероятности:

$$
P(A) = P(A|H1)P(H1) + P(A|H2)P(H2) + P(A|H3)P(H3) = {1 \over 17} \times 0.8 + {4 \over 17} \times 0.9 + {12 \over 17} \times 0.85 = 0.86
$$

## Задание 3

Из 40 снайперов 18 попадает в цель с вероятностью 0,9, 8 — с вероятностью 0,4 и 14 – с вероятностью 0,7. Случайно выбранный снайпер произвел выстрел, поразив цель. Мы хотим вычислить, снайпер из какой группы наиболее вероятно совершил этот выстрел. 

В ответ запишите наибольшую вероятность, округленную до тысячных. В качестве разделителя используйте точку.

Подсказки

- **Ну а тут как?**
    
    Здесь мы наконец-то используем формулу Байеса целиком:
    
    $$
    P(Y|X) = {P(Y) P(X|Y) \over P(X)}
    $$
    
    $P(X)$ вычислять мы уже научились на материале прошлых задачек. Теперь вам нужно вычислить $P(Y|X)$ для каждой из трёх групп снайперов и выбрать ту, которая больше всего.

In [ ]:
40 всего
18 - 0.9 а
8  - 0.4 б
14 - 0.7 с


In [9]:
# Общая вероятность попадания
Pa = 18/40*0.9 + 8/40*0.4 + 14/40*0.7
Pa

0.73

In [10]:
#вероятность для группы А
18/40*0.9/Pa 

0.5547945205479452

In [ ]:
#Вероятность для группы Б

In [11]:
8/40*0.4/Pa

0.10958904109589043

In [12]:
#Вероятность для группы С
14/40/Pa

0.4794520547945205

In [13]:
#самая высокая вероятность у группы А
round(18/40*0.9/Pa, 3)

0.555

##### ОТВЕТ: 0.555 с первой попытки без подсказок

**Разбор решения**

Cделаем следующее допущение: 

H1 - снайпер принадлежал первой группе 

H2 - снайпер принадлежал второй группе 

H3 - снайпер принадлежал третьей группе

Тогда мы можем по классическому определению вероятности рассчитать:

$$
P(H1) = {18 \over 40} = 0.45
$$

$$
P(H2) = {14 \over 40} = 0.35
$$

$$
P(H3) = {8 \over 40} = 0.2
$$

Найдем вероятность события А(снайпер попал в мишень) и введем условные вероятности:

$$
P(A|H1) = 0.9
$$

$$
P(A|H2) = 0.7
$$

$$
P(A|H3) = 0.4
$$

И тогда по формуле полной вероятности:

$$
P(A) = P(A|H1) \times P(H1) + P(A|H2) \times P(H2) + P(A|H3) \times P(H3) = 0.45 \times 0.9 + 0.35 \times 0.7 + 0.2 \times 0.4 = 0.73
$$

Теперь найдем апостериорные вероятности того, что стрелок принадлежит к конкретной группе, если он попал в цель:
$$
**Разбор решения**

Cделаем следующее допущение: 

H1 - снайпер принадлежал первой группе 

H2 - снайпер принадлежал второй группе 

H3 - снайпер принадлежал третьей группе

Тогда мы можем по классическому определению вероятности рассчитать:

$$
P(H1) = {18 \over 40} = 0.45
$$

$$
P(H2) = {14 \over 40} = 0.35
$$

$$
P(H3) = {8 \over 40} = 0.2
$$

Найдем вероятность события А(снайпер попал в мишень) и введем условные вероятности:

$$
P(A|H1) = 0.9
$$

$$
P(A|H2) = 0.7
$$

$$
P(A|H3) = 0.4
$$

И тогда по формуле полной вероятности:

$$
P(A) = P(A|H1) \times P(H1) + P(A|H2) \times P(H2) + P(A|H3) \times P(H3) = 0.45 \times 0.9 + 0.35 \times 0.7 + 0.2 \times 0.4 = 0.73
$$

Теперь найдем апостериорные вероятности того, что стрелок принадлежит к конкретной группе, если он попал в цель:
$$
P(H1|A) = {P(A|H1) \times P(H1) \over P(A)} = {0.45 \times 0.9 \over 0.73} = 0.55479
$$
$$
P(H2|A) = {P(A|H2) \times P(H2) \over P(A)} = {0.35 \times 0.7 \over 0.73} = 0.3356
$$
$$
P(H3|A) = {P(A|H3) \times P(H3) \over P(A)} = {0.2 \times 0.4 \over 0.73} = 0.10959
$$

## Задание 4

Есть 4 кубика. На трех из них окрашена белым половина граней, а на четвертом кубике всего одна грань из шести белая. Наудачу выбранный кубик подбрасывается семь раз. Найти вероятность того, что был выбран четвертый кубик, если при семи подбрасываниях белая грань выпала ровно один раз.  Ответ округлите до тысячных. Используйте точку в качестве десятичного разделителя.

Подсказки

- **А что делать тут?**
    
    А здесь нам понадобится вспомнить формулу Бернулли!
    
    $$
    P_{n}^{k}=C_{n}^{k}p^{k}q^{n-k}
    $$
    
    Вероятность выкинуть 1 белую грань из 7 считаем по ней для каждого из двух типов кубиков, а дальше пользуемся формулой Байеса, как в прошлой задаче. Не забудьте подсчитать вероятность взять конкретный кубик!

In [26]:
from scipy import special #модуль питона, предназначенный для научных и инженерных расчетов

# special.comb(n, k) #биномиальный коэффициент



#альтернативный вариант

from math import factorial

def num_of_successes(n, k):
    return factorial(n)/(factorial(k) * factorial(n - k))

In [ ]:
4 кубика
3 3 белых(грани) 3 другие(грани)
1 1 белая(грань) 5 других(грраней)

кидаем 7 раз



In [14]:
#Вероятность кубика с 3 гранями 
3/4

0.75

In [16]:
#вероятность что белая грань 
3/6

0.5

In [15]:
#Вероятность кубика с 1 гранью
1/4

0.25

In [37]:
#вероятность что белая грань 
1/6

0.16666666666666666

In [36]:
#вероятность что белая грань 7 раз для 1 го кубика
1*2*3*4*5*6/(1 * 1*2*3*4*5)

6.0

In [19]:
(1*2*3)/(1*2*1)


3.0

In [51]:
#Вероятность 1 раз из 7 для кубика №2
special.comb(7, 1)*0.16666666666666666**1*(1-0.16666666666666666)**6

0.39071430612711483

In [42]:
#Вероятность 1 раз из 7 для кубика №1
special.comb(7, 1)*0.5**1*0.5**6

0.0546875

In [44]:
#Общая вероятность:

In [53]:
0.25* 0.39071430612711483 + 0.75*0.0546875

0.1386942015317787

In [56]:
# вероятность что был всетаки второй кубик с 1 гранью
(1/4* 0.39071430612711483)/0.1386942015317787

0.7042729649328406

In [55]:
round(0.7042729649328406, 3)

0.704

###### ОТВЕТ: 0.704 с 3й попытки и с  подсказкой(запутался с подстановками)

**Разбор решения**

Пусть H1 это событие, когда выбирают 1, 2 или третий кубик, а H2 - когда выбирают 4-ый кубик.

Тогда по классическому определению вероятности: 

$$
P(H1) = {3 \over 4}, \ P(H2) = {1 \over 4}
$$

Тогда для события H1 вероятность появления белой грани p = 0.5. 

Посчитаем вероятность того, что при n = 7 (семи бросках) белая грань выпадает ровно k = 1. Используем формулу Бернулли:

$$
P_{7}^{1}=C_{7}^{1}0.5^{1}0.5^{7-1} = 0.0547 = P(A|H1)
$$

Теперь рассматриваем событие H2

Вероятность появления белой грани 

$$
p = \frac{1}{6}
$$

тогда вероятность по формуле Бернулли при n = 7, а k = 1 будет:

$$
P_{7}^{1}=C_{7}^{1}{1 \over 6}^{1}{5 \over 6}^{7-1} = 0.3907 = P(A|H2)
$$

Тогда по формуле полной вероятности получаем:

$$
P(A) = P(H1) \times P(A|H1) + P(H2) \times P(A|H2) = 0.75 \times 0.0547 + 0.25 \times 0.3907 = 0.1387
$$

И нам остался финальный штрих - уточнить вероятность гипотезы H2

$$
P(H2|A) = {P(A|H2) \times P(H2) \over P(A)} = {0.25 \times 0.3907 \over 0.1387} = 0.7042
$$

## Задание 5

На гитхабе лежит 12 скриптов, написанных аналитиком №1, 20 скриптов — написанных аналитиком №2 и 18 скриптов — написанных аналитиком №3. Вероятность того, что скрипт, написанный аналитиком №1, работает  без ошибок, равна 0,9; для скриптов, написанных аналитиками  №2 и №3, эти вероятности соответственно равны 0,6 и 0,9.

Нужно найти вероятность того, что случайно взятый скрипт будет работать без ошибок. В ответ запишите число, округленное до сотых. В качестве десятичного разделителя используйте точку.

Подсказки

- **Это похоже на первую задачку в уроке или мне кажется?**
    
    Не кажется! Решается всё так же, через вычисление $P(X)$.

In [ ]:
№1 12 0.9
№2 20 0.6
№3 18 0.9
 всего 50 скриптов



In [57]:
# # Не верно!!!(()) потому что  20+18+12 = 50
# 12/40 * 0.9 + 20/40 * 0.6 + 18/40 * 0.9# Не верно!!!(())

0.9750000000000001

In [58]:
# # Не верно!!!(())
# round(0.9750000000000001, 2)

0.98

In [59]:
# ЭТО тоже бред
# 12*0.9**1*0.1**11 + 20*0.6**1*0.4**19 + 18*0.9**1*0.1**17

3.2996148849480035e-07

In [61]:
# # ЭТО тоже бред
# round(3.2996148849480035e-07, 3)

0.0

In [62]:
# После полсказки
12/40*0.9 + 20/40*0.6 + 18/40*0.9

0.9750000000000001

#### Я ТУПО НЕПРАВИЛЬНО СЛОЖИЛ !!!! ПЕРВОЕ РЕШЕНИЕ ВЕРНО !!! ВСЕГО 50 скриптов а не 40

In [63]:
round((12/50 * 0.9 + 20/50 * 0.6 + 18/50 * 0.9),2)

0.78

##### ОТВЕТ: 0.78 со второй попытки, и подсказка не помогла и не нужна была тупо сумму изначально неправильно посчитал

**Разбор решения**

Давайте найдем вероятность каждого события по классическому определению вероятности. Всего 50 скриптов, из них первый аналитик написал 12, то есть 

$$
P(H1) = {12 \over 50} = 0.24
$$

2-ой аналитик написал 20, то есть 

$$
P(H2) = {20 \over 50} = 0.4, 
$$

остальные скрипты написал третий аналитик, то есть 

$$
P(H3) = {18 \over 50} = 0.36.
$$

Введем событие А = (скрипт без ошибок). По условию даны априорные вероятности 

$$
P(A|H1) = 0.9, P(A|H2) = 0.6, P(A|H3) = 0.9
$$

Вероятность события А можем найти по формуле полной вероятности: 

$$
P(A) = P(A|H1)P(H1) + P(A|H2)P(H2) + P(A|H3)P(H3) = 0.9 \times 0.24 + 0.6 \times 0.4 + 0.9 \times 0.36 = 0.78
$$

## Задание 6

Известно, что около 60% всех людей в городе болеют заболеванием А. Для определения того, заболел человек или нет, проводится тест, точность которого составляет 99%. Вы получили положительный тест, какова вероятность того, что вы реально больны?

Округлите ответ до тысячных, в качестве десятичного раделителя используйте точку.

Подсказки

- **Тут реализуем формулу Байеса, так?**
    
    Так! Смотрим, что нам надо посчитать:
    
    - $P(Y|X)$ - вероятность, что мы реально больны, если тест положительный (**то, что мы ищем в задаче**)
    - $P(X|Y)$ - вероятность, что тест будет положительный, если мы заболеем (**точность теста, указана в условии**)
    - $P(Y)$ - вероятность заболеть (**также указана в условии**)
    - $P(X)$ - вероятность того, что тест положительный (**считаем по аналогии с прошлыми задачами**)
    
- **Окей, посчитали, всё равно не работает!**
    
    А ещё убедитесь, что вы указываете результат именно в десятичной записи, а не в процентах :)

In [ ]:
точность 0.99
болеют 0,6
не болеют 0,4


In [64]:
# не верно
# # ОБщая верооятность положительного теста
# 0.6*0.99 + 0.4*0.99

0.99

In [68]:
0.6*0.99/(0.6*0.99 + 0.4*0.01)

0.9933110367892977

In [69]:
round(0.9933110367892977, 3)

0.993

##### ОТВЕТ 0.993 со второй попытки и с помощью подсказки

# Задание 6

**Разбор решения**

Пусть,

A - событие, которое соответствует болезни человека ,

H_1 - событие, которое соответствует положительному результату теста

H_2 - событие, которое соответствует отрицательному результату теста

Тогда по формуле Байеса,

$$
P(A|H_1) = \frac{P(H_1|A) * P(A)}{P(H_1)}
$$

$$
P(H_1|A) = 0.99, 
$$

так как мы знаем, что если человек болеет, то тест это покажет с 99 % точностью

$$
P(A) = 0.6 - вероятность \ того, что \ человек \ болен
$$

По формуле полной вероятности,

$$
P(H_1) = P(H_1|A) * P(A) + P(H_1|!A) * P(!A) = 0.99 * 0.6 + 0.01 * 0.4 = 0.598
$$

Подставляем всё в формулу:
$$
P(A|H_1) = \frac{0.99 * 0.6}{0.598} ≈ 0.993
$$

## Задание 7

Три фабрики производят лампочки. Фабрика А производит 20% всех лампочек, 50% лампочек производится на фабрике В и оставшиеся 30% на фабрике С. 2% лампочек, произведенных на фабрике А, 1% лампочек, произведенных на фабрике В, и 3% лампочек, произведенных на фабрике С, являются бракованными.На рынке наугад выбирается лампочка и оказывается бракованной. Какова вероятность того, что эта лампочка была произведена на заводе B?

Ответ округлите до тысячных, в качестве разделителя  используйте точку.

Подсказки

- **И тут байес?**
    
    И тут байес. Смотрим:
    
    - $P(Y|X)$ - вероятность, что **именно эту** бракованную лампочку сделали на заводе B (**то, что мы ищем в задаче**)
    - $P(X|Y)$ - вероятность, что завод B в принципе сделает бракованную лампочку (**указана в условии**)
    - $P(Y)$ - вероятность того, что лампочка (любая) сделана на заводе B (**также указана в условии**)
    - $P(X)$ - вероятность найти бракованную лампочку из всех возможных вариантов (**считаем по аналогии с прошлыми задачами, учитываем все три завода с их вероятностями брака и количеством продукции**)

In [ ]:
        брак
a 0.2 0.02
b 0.5 0.01
c 0.3 0.03

In [70]:
# Общая вероятность события брака 
0.2*0.02 + 0.5*0.01 + 0.3*0.03

0.018000000000000002

In [72]:
# вероятность что это B
0.5*0.01 /(0.2*0.02 + 0.5*0.01 + 0.3*0.03)

0.27777777777777773

In [73]:
round(0.27777777777777773, 3)

0.278

###### ОТВЕТ: 0.278 с первой попытки без подсказок

**Разбор решения**

Пусть, A, B, C -  события, соответствующие тому, что лампочка с завода A, B и C соответственно, а H - событие соответствующее тому, что лампочка бракованная, тогда
$$
P(B|H) = \frac{P(H|B) P(B)}{P(H)}
$$
$$
P(H|B) = 0.01
$$
$$
P(B) = 0.5
$$
$$
P(H) = P(H|A) * P(A) + P(H|B) * P(B) + P(H|C) * P(C) =0.02 * 0.2 + 0.01 * 0.5 + 0.03 * 0.3 = 0.018
$$
$$
P(B|H) = \frac{0.01 * 0.5}{0.018} ≈ 0.278
$$


## Задание 8

Система РЛС (радиолокационная система) может обнаружить присутствие самолета в 98% случаев. Однако если в зоне действия радара нет самолета, система все равно сообщает (ложно), что самолет присутствует, в 5% случаев. В любой момент времени вероятность того, что самолет находится в зоне действия радара, составляет 7%.Какова вероятность того, что самолет находится в зоне действия радара, учитывая, что на радаре нет самолета?

Ответ округлите до десятитысячных, в качестве десятичного разделителя используйте точку.

Подсказки

- **Удивительно, как много задач в уроке про формулу Байеса касается формулы Байеса.**
    
    Да сам удивляюсь, не иначе звёзды сошлись.
    
    - $P(Y|X)$ - вероятность, что самолёт есть, если радар его не видит (**то, что мы ищем в задаче**)
    - $P(X|Y)$ - вероятность, что радар ничего не увидит, если самолёт прилетит (**в условии указана обратная вероятность - что радар увидит реально летящий самолёт**)
    - $P(Y)$ - вероятность, что самолёт в принципе может залететь в область радара (**также указана в условии**)
    - $P(X)$ - вероятность того, что самолёт не будет видно на радаре в принципе (**считаем по аналогии с прошлыми задачами, учитываем указанную в условии вероятность ложноположительного срабатывания, но про ложноотрицательное тоже не забывайте**)

In [ ]:
0.93 НЕТ 0.07 есть
погрешность 0.98 обнаружения
0.05 погрешность если нету самолета


In [109]:
# Не верно
# # Общая вероятность что обнаружится??
# 0.93*0.98 + 0.07*0.05

In [111]:
# # не верно
# round((1- 0.93*0.98 + 0.07*0.05), 4)

In [110]:
# Не верно
# round((0.93*0.02/(0.93*0.98 + 0.07*0.05)), 4)

In [113]:
# # Не верно
# round((0.93*0.02/(0.93*0.02 + 0.07*0.95)), 4)

In [85]:
# Не верно
# round((0.93*0.02 /(0.93*0.02 + 0.07*0.95)), 4)

0.2186

In [108]:
#ВЕРНОЕ
# Веротяность что не обнаружится
# самолет есть но не обнаружился + самолета  нет
0.07*0.02 + 0.93*0.95

0.8848999999999999

In [107]:
round(( 0.07*0.02 /(0.07*0.02 + 0.93*0.95 )), 4)

0.0016

##### ОТВЕТ: 0.0016 с 15 попытки Внимательнее условие надо читать

**Разбор решения**

Система РЛС (радиолокационная система) может обнаружить присутствие самолета в 98% случаев. Однако если в зоне действия радара нет самолета, система все равно сообщает (ложно), что самолет присутствует, в 5% случаев. В любой момент времени вероятность того, что самолет находится в зоне действия радара, составляет 7%. Какова вероятность того, что самолет находится в зоне действия радара, учитывая, что на радаре нет самолета?

**Разбор решения**

A - событие, соответствующее тому, что самолет в зоне

B - событие, соответствующее тому, что самолет не обнаружен

Тогда вероятность того, что самолет не будет обнаружен, даже если он в зоне:

$$
P(B|A) = 1 - 0.98 = 0.02
$$

А вероятность того, что самолет не обнаружат, если его реально нет в зоне:

$$
P(B|A`) = 1 - 0.05 = 0.95
$$

Воспользуемся теоремой Байеса:

$$
P(A|B) = \frac {P(B|A)*P(A)} {P(B|A)*P(A) + P(B|A`)*P(A`)} = \frac {0.02 * 0.07} {0.02*0.07 + 0.95*0.93} = 0.0016
$$

## Задание 9

Система РЛС (радиолокационная система) может обнаружить присутствие самолета в 98% случаев. Однако если в зоне действия радара нет самолета, система все равно сообщает (ложно), что самолет присутствует, в 5% случаев. В любой момент времени вероятность того, что самолет находится в зоне действия радара, составляет 7%.Какова вероятность того, что самолет находится в зоне действия радара, учитывая, что радар показывает наличие самолета?

Ответ округлите до тысячных, в качестве десятичного разделителя используйте точку.

Подсказки

- **Здесь что-то надо поменять, да?**
    
    Ага, всего два элемента:
    
    - $P(X|Y)$ - здесь уже берём именно вероятность того, что самолёт будет замечен радаром
    - $P(X)$ - здесь считаем вероятность положительного срабатывания радара (как верного, так и ложного)

In [ ]:
0.93 НЕТ 0.07 есть
погрешность 0.98 обнаружения
0.05 погрешность если нету самолета


In [114]:
#Вероятносьть что радар показывает можно тупо отпредыдущего задания взять 1-0.07*0.02 + 0.93*0.95
0.07*0.98 + 0.93*0.05

0.11510000000000001

In [115]:
round((0.07*0.98/(0.07*0.98 + 0.93*0.05)), 3)

0.596

##### ОТВЕТ: 0.596 с 1 попытки и без подсказок( потому что на предыдущей натрахался)

**Разбор решения**

В принципе, все происходит по аналогии с задачей №8.

A - событие, соответствующее тому, что самолет **в зоне**

B - событие, соответствующее тому, что самолет **обнаружен**

Тогда вероятность того, что самолет будет обнаружен, если он находится в зоне:

$$
P(B|A) = 0.98
$$

А вероятность того, что самолет обнаружат, даже если его в зоне нет:

$$
P(B|A`) = 0.05
$$

Воспользуемся теоремой Байеса:
$$
P(A|B) = \frac {P(B|A)P(A)} {P(B|A)P(A) + P(B|A`)*P(A`)} = \frac {0.98 * 0.07} {0.980.07 + 0.050.93} = 0.596
$$

## Задание 10

Передатчик аварийного локатора самолета (ELT) - это устройство, предназначенное для передачи сигнала в случае аварии. Компания A производит 80% всех ELT на рынке, компания B - 15%, а компания C - остальные 5%. Уровень дефектности ELT, произведенных компанией A, составляет 4%, компанией B - 6%, а компанией C - 9%. Случайно выбранный ELT был протестирован и оказался бракованным. Найдите вероятность того, что он был произведен компанией A.

Округлите ответ до тысячных, в качестве десятичного разделителя используйте точку.

Подсказки

- **Это как та задачка про лампочки?**
    
    Всё так, всё так! На всякий случай снова распишем все элементы формулы:
    
    - $P(Y|X)$ - вероятность, что этот бракованный ELT был сделан компанией A (**то, что мы ищем в задаче**)
    - $P(X|Y)$ - вероятность, что если ELT делается компанией A, то он будет бракованный (**указана в условии**)
    - $P(Y)$ - вероятность того, что любой ELT сделан именно компанией A (**также указана в условии**)
    - $P(X)$ - вероятность бракованного ELT в принципе (**считаем по аналогии с прошлыми задачами, учитываем все три компании**)

In [ ]:
0.8   0.04
0.15  0.06
0.05  0.09

In [116]:
# Общая вероятность брака

0.8*0.04 + 0.15*0.06 + 0.05*0.09

0.0455

In [117]:
round((0.8*0.04/(0.8*0.04 + 0.15*0.06 + 0.05*0.09)), 3)

0.703

##### ОТВЕТ:0.703 с 1 попытки и без подсказок

**Разбор решения**

Пусть, A, B, C -  события, соответствующие тому, что ELT от компании A, B и C соответственно, а H - событие соответствующее тому, что ELT бракованный, тогда

$$
P(A|H) = \frac{P(H|A) P(A)}{P(H)}
$$

$$
P(H|A) = 0.04
$$

$$
P(A) = 0.8
$$

$$
P(H) = P(H|A) * P(A) + P(H|B) * P(B) + P(H|C) * P(C) =0.04 * 0.8 + 0.06 * 0.15 + 0.09 * 0.05 = 0.0455
$$

$$
P(A|H) = \frac{0.04 * 0.8}{0.0455} ≈ 0.703
$$